src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/pipeline.js

```diff
@@ -108,24 +107,44 @@ export default class Pipeline {
     return result;
   }
 
-  // Updates the state of the pipeline and processors with the results
-  // from an ingest simulate call.
-  applySimulateResults(results) {
-    //update the outputObject of each processor
-    results.forEach((result) => {
+  _updateProcessorOutputs(simulateResults) {
+    simulateResults.forEach((result) => {
       const processor = this.getProcessorById(result.processorId);
 
       processor.outputObject = _.get(result, 'output');
       processor.error = _.get(result, 'error');
     });
+  }
 
-    //update the inputObject of each processor
-    results.forEach((result) => {
-      const processor = this.getProcessorById(result.processorId);
+  //Updates the error state of the pipeline and its processors
+  //If a pipeline compile error is returned, lock all processors but the error
+  //If a pipeline data error is returned, lock all processors after the error
+  _updateErrorState() {
+    this.hasCompileError = _.some(this.processors, (processor) => {
+      return _.get(processor, 'error.compile');
+    });
+    _.forEach(this.processors, processor => {
+      processor.locked = false;
+    });
 
+    const errorIndex = _.findIndex(this.processors, 'error');
+    if (errorIndex === -1) return;
+
+    _.forEach(this.processors, (processor, index) => {
+      if (this.hasCompileError && index !== errorIndex) {
+        processor.locked = true;
```

Comment:
I don't think we ever want a processor with a compile error to be locked. This makes it possible for a user to get into a situation where they have a compile error that they can't fix because the processor is locked. I can't figure out how to reproduce it, but I accidentally got into this state once while testing, and I suspect this line is the culprit.